crates/valence_network/src/lib.rs

```diff
@@ -549,3 +576,31 @@ pub struct PlayerSampleEntry {
     /// The player UUID.
     pub id: Uuid,
 }
+
+async fn do_broadcast_to_lan_loop(shared: SharedNetworkState) {
+    let port = shared.0.address.port();
+
+    // connect to server_addr on Udp to send a text packet
+    // if this ever fails, you have bigger problems to worry about, since we're
+    // asing the OS for a port
+    let socket = tokio::net::UdpSocket::bind("0.0.0.0:0").await.unwrap();
```

Comment:
Even in this scenario I think it's a bad idea to panic. Better to log an error message and return.